OverlayPlugin.Core/EventSources/EnmityEventSource.cs

```diff
@@ -97,20 +95,24 @@ public override void SaveConfig(IPluginConfig config)
         {
         }
 
-        private void UpdateInCombat()
+        private void OnInCombatChanged(object sender, InCombatArgs args)
         {
-            if (!inCombatMemory.IsValid())
-                return;
+            var combatData = new InCombatDataObject();
+            combatData.inACTCombat = args.InACTCombat;
+            combatData.inGameCombat = args.InGameCombat;
+            DispatchAndCacheEvent(JObject.FromObject(combatData));
 
-            // Handle optional "end encounter of combat" logic.
-            bool inGameCombat = inCombatMemory.GetInCombat();
-            if (inGameCombat != lastInGameCombat)
+            if (!args.InGameCombatChanged)
             {
-                logger.Log(LogLevel.Debug, inGameCombat ? "Entered combat" : "Left combat");
+                return;
             }
 
+            // Handle optional "end encounter of combat" logic.
+            bool inGameCombat = args.InGameCombat;
+            logger.Log(LogLevel.Debug, inGameCombat ? "Entered combat" : "Left combat");
+
             // If we've transitioned to being out of combat, start a delayed task to end the ACT encounter.
-            if (Config.EndEncounterOutOfCombat && lastInGameCombat && !inGameCombat)
+            if (Config.EndEncounterOutOfCombat && !inGameCombat)
```

Comment:
Should this check for an existing token as well to avoid spawning multiple tokens and tasks?

Reply:
The only way to spawn an additional task is to transition into combat and then out of it again.  If you transition into combat, then it will cancel.  So, I think no need to additionally check for cancelling when transitioning out of combat.